### ISSUE TITLE
Install halts after reboot when installing 2 packages

### ISSUE THREAD

**Issue by riezebosch** [mwrock/boxstarter#163]
This is my script: 
```powershell
New-PackageFromScript "https://gist.githubusercontent.com/riezebosch/6b0523a19f6b5eafcc9f/raw/_install.ps1" BOXSTARTER_VS2015
New-PackageFromScript "https://gist.githubusercontent.com/riezebosch/6b0523a19f6b5eafcc9f/raw/shutdown.ps1" BOXSTARTER_SHUTDOWN
Install-BoxstarterPackage BOXSTARTER_VS2015,BOXSTARTER_SHUTDOWN -Credential $cred
```

I first generate 2 packages from 2 different gists and then invoke the Install-BoxstarterPackage with both packages. This is because I reuse the shutdown script in several different installations.

In version 2.6 this works fine, but since 2.7 it fails the continue after a reboot that is invoked from within the first package.

![boxstarter_after_reboot](https://cloud.githubusercontent.com/assets/1427334/14278352/860b0e4c-fb27-11e5-9703-7ffdf3806ae8.PNG)

Here's the last portion of the log file:
```
Chocolatey installed 1/1 package(s). 0 package(s) failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
[2016-04-05T11:29:49.6003188+02:00:::PID 2000] Boxstarter: restoring current directory location to C:\windows\system32
[2016-04-05T11:29:49.6003188+02:00:::PID 2000] Boxstarter: found C:\Users\Administrator\AppData\Local\Boxstarter\Boxstarter.2000.restart we are restarting
[2016-04-05T11:29:49.6003188+02:00:::PID 2000] Boxstarter: found C:\Users\Administrator\AppData\Local\Boxstarter\Boxstarter.2000.restart we are restarting
[2016-04-05T11:29:49.6315691+02:00:::PID 2000] Boxstarter: Exit Code: 0
[2016-04-05T11:29:49.6784444+02:00:::PID 2000] + Boxstarter finished Calling Chocolatey to install BOXSTARTER_VS2015. This may take several minutes to complete... 00:12:07.4394493
[2016-04-05T11:29:49.6784444+02:00:::PID 2000] + Boxstarter finished Calling Chocolatey to install BOXSTARTER_VS2015. This may take several minutes to complete... 00:12:07.4394493
[2016-04-05T11:29:49.6940702+02:00:::PID 2000] Boxstarter: Removing C:\ProgramData\chocolatey\lib\BOXSTARTER_VS2015 in progress
[2016-04-05T11:29:49.7096954+02:00:::PID 2000] Boxstarter: writing restart file
[2016-04-05T11:29:49.7096954+02:00:::PID 2000] Boxstarter: writing restart file
[2016-04-05T11:29:49.7721911+02:00:::PID 2000] Boxstarter: Restore Automatic Updates from Windows Update
[2016-04-05T11:29:49.8190675+02:00:::PID 2000] Boxstarter: UAC Enabled. Disabling...
[2016-04-05T11:29:49.8346915+02:00:::PID 2000] Boxstarter: Disabling UAC
[2016-04-05T11:29:49.8971917+02:00:::PID 2000] Boxstarter: Securely Storing DOCENTA\Administrator credentials for automatic logon
[2016-04-05T11:29:50.5394990+02:00:::PID 2000] Boxstarter: Logon Set
[2016-04-05T11:29:50.5707503+02:00:::PID 2000] + Boxstarter finished Installation session. 00:12:09.2844118
[2016-04-05T11:29:50.5707503+02:00:::PID 2000] Boxstarter: Restart Required. Restarting now...
[2016-04-05T11:29:50.9948486+02:00:::PID 2000] Boxstarter: Composing record for pipeline...
[2016-04-05T11:29:50.9948486+02:00:::PID 2000] Boxstarter: writing object...
[2016-04-05T11:29:51.0260918+02:00:::PID 2000] Boxstarter: object written...
[2016-04-05T11:33:36.7027209+02:00:::PID 3020] parent process is cmd
[2016-04-05T11:33:36.7808272+02:00:::PID 3020] parent process is explorer
[2016-04-05T11:33:36.9527297+02:00:::PID 3020] Error getting parent process
[2016-04-05T11:33:37.1558273+02:00:::PID 3020] Boxstarter Version 2.7.0
[2016-04-05T11:33:37.1714525+02:00:::PID 3020] (c) 2015 Matt Wrock. http://boxstarter.org

[2016-04-05T11:33:37.2027029+02:00:::PID 3020] + Boxstarter starting Installation session.
[2016-04-05T11:33:37.2183278+02:00:::PID 3020] Boxstarter: NoPassword is false checking autologin
[2016-04-05T11:33:37.2808277+02:00:::PID 3020] Boxstarter: AutoLogin status is True
[2016-04-05T11:33:37.3120778+02:00:::PID 3020] Boxstarter: NoPassword is set to True and RebootOk is set to True and the NoPassword parameter passed was False
[2016-04-05T11:33:37.3277029+02:00:::PID 3020] Boxstarter: Disabling Automatic Updates from Windows Update
[2016-04-05T11:33:38.2652031+02:00:::PID 3020] Boxstarter: LocalRepo is at C:\VPC_images
[2016-04-05T11:33:38.3120787+02:00:::PID 3020] Boxstarter: Installing 2 packages from C:\VPC_images;http://chocolatey.org/api/v2;http://www.myget.org/F/boxstarter/api/v2
[2016-04-05T11:33:38.3745785+02:00:::PID 3020] You cannot call a method on a null-valued expression.
[2016-04-05T11:33:38.3745785+02:00:::PID 3020] Boxstarter: Restore Automatic Updates from Windows Update
[2016-04-05T11:33:38.5464538+02:00:::PID 3020] Boxstarter: Enabling UAC
[2016-04-05T11:33:38.5777042+02:00:::PID 3020] Boxstarter: Cleaning up and not rebooting
[2016-04-05T11:33:38.5777042+02:00:::PID 3020] Boxstarter: Cleaning up restart file
[2016-04-05T11:33:38.5777042+02:00:::PID 3020] Boxstarter: Cleaning up autologon registry keys
```

I tried invoking boxstarter with the -verbose parameter, but this also seems to be lost after a reboot.

**Comment by mwrock** [comment]
@levande and @riezebosch I have been having a hard time reproducing this. A few questions:
1. What Windows version are you running the package on?
2. Do you get this error by running any 2 packages or just the ones you mentioned in your issues? For instance, do you get the error running `7zip,curl`? @riezebosch mentioned this happens after a reboot but I wonder if it also happens without a reboot.

Thanks!

**Comment by mwrock** [comment]
ok @riezebosch an @levande I'm almost nearly certain I have a repro and a fix here. It would still be great if you have answers to the above. While I believe I have a fix, I also want to determine why you both got such horrible errors with just the message and no stack trace.

**Comment by levande** [comment]
Hi Matt

First I want to say thank you for assisting on this matter. I really like your module!

I´m running on Windows 8.1. I have only tried to run my own made package. This worked perfectly before I upgraded to latest version. Before I used 2.5.21.

I now tried to trigger a new installation with 7zip and curl and the problem seems to be the same,

[cid:image001.png@01D193CC.539796F0]

Regards
Jimmy
